### PR TITLE
Improve Javadoc.

### PR DESCRIPTION
**Pull request by arturobernalg**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [X ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

<!--
This pull request improves the Javadoc for several methods in the Liquibase project. The changes include clarifying method descriptions, adding parameter and return value information, and deprecating some methods with alternative suggestions. These updates aim to make the codebase more readable and user-friendly.

### CODE REVIEW

**Comment by nvoxland** [comment]
**Initial/Pre-Review Thoughts**

I didn't actually read through the javadoc changes, so if there's questions during the main review feel free to ask.

Questions I have:
- None

Potential risks:
- None

What could make the full review difficult:
- Nothing

**Comment by arturobernalg** [comment]
> **Initial/Pre-Review Thoughts**
> 
> I didn't actually read through the javadoc changes, so if there's questions during the main review feel free to ask.
> 
> Questions I have:
> 
> * None
> 
> Potential risks:
> 
> * None
> 
> What could make the full review difficult:
> 
> * Nothing

I did.  Just improvement and add new one.

**Review by MalloD12 — APPROVED** [review]
**Approved**.

## Review and testing results:

This looks ok to me. I have only made a very few minor tweaks to some javadocs. Thanks @arturobernalg. 

### Things to be aware of:
- None

### Things to worry about:
- None

**Review by filipelautert — APPROVED** [review]

File: liquibase-standard/src/main/java/liquibase/hub/core/StandardHubService.java [hunk 1]
```diff
@@ -357,12 +357,12 @@ public String shortenLink(String url) throws LiquibaseException {
     }
 
     /**
-     * Query for a changelog ID.  If no result we return null
-     * We cache this result and a map
+     * Queries for a changelog ID. If no result is found, it returns {@code null}.
```
Comment (XDelphiGrl) [inline liquibase-standard/src/main/java/liquibase/hub/core/StandardHubService.java:360]: This whole class is going away with Hub decommission. I bet we see that show up soon now that the Hub decommission branch is merged to master ...

File: liquibase-standard/src/main/java/liquibase/statement/ExecutablePreparedStatement.java [hunk 2]
```diff
@@ -8,9 +8,10 @@
  */
 public interface ExecutablePreparedStatement extends SqlStatement {
     /**
-     * Execute the prepared statement
-     * @param factory for creating a <code>PreparedStatement</code> object
-     * @throws DatabaseException
+     * Executes the prepared statement created by the given factory.
+     *
+     * @param factory a factory for creating a <code>PreparedStatement</code> object.
```
Comment (XDelphiGrl) [inline liquibase-standard/src/main/java/liquibase/statement/ExecutablePreparedStatement.java:13]: Should `<code>PreparedStatement</code>` be `{@code PreparedStatement}'?

File: liquibase-standard/src/main/java/liquibase/util/StringUtil.java [hunk 3]
```diff
@@ -593,11 +593,14 @@ public static boolean isNotEmpty(String value) {
...
-     * Checks if <code>value</code> starts with <code>startsWith</code>.
+     * Checks whether the given <code>value</code> starts with the specified <code>startsWith</code> string.
      *
-     * @param value
-     * @param startsWith
-     * @return true if <code>value</code> starts with <code>startsWith</code>, otherwise false. If any of arguments is null returns false
+     * @param value      the string to check
+     * @param startsWith the prefix to check for
+     * @return <code>true</code> if <code>value</code> starts with <code>startsWith</code>, <code>false</code> otherwise.
+     * css
```
Comment (XDelphiGrl) [inline liquibase-standard/src/main/java/liquibase/util/StringUtil.java:601]: css and Copy code seem sort of random. Typos??

And, should the `<code></code>` blocks be `@code` instead?

**Review by XDelphiGrl — CHANGES_REQUESTED** [review]
I've left a few comments for (potential) requested changes.

I also validated that we're not making any changes to serialization in JavaDoc annotations to prevent regressions like [https://github.com/liquibase/liquibase/pull/4161/files](https://github.com/liquibase/liquibase/pull/4161/files).

File: liquibase-standard/src/main/java/liquibase/Liquibase.java [hunk 4]
```diff
@@ -359,16 +360,17 @@ protected RollbackVisitor createRollbackVisitor(List<ChangesetsRolledback.Change
...
-     * @param   contexts                           Contexts to filter for
-     * @param   labelExpression                    Labels to filter for
-     * @param   changeLog                          The changelog to process
-     *
-     * @return  ChangeLogIterator
-     * @throws DatabaseException
-     *
+     * @param contexts          The contexts to filter for.
+     * @param labelExpression   The labels to filter for.
+     * @param collectAllReasons A flag to control whether all skip reasons are accumulated.
```
[the remark this answers is not in the document]
Comment (MalloD12) [reply liquibase-standard/src/main/java/liquibase/Liquibase.java:367]: Hey Erz, I'm going to update this as I saw it collectAllReasons wasn't there previously:
```
/**
     *
     * Return a ChangeLogIterator constructed with standard filters
     *
     * @param   contexts                           Contexts to filter for
     * @param   labelExpression                    Labels to filter for
     * @param   changeLog                          The changelog to process
     *
     * @return  ChangeLogIterator
     * @throws DatabaseException
     *
     */
    protected ChangeLogIterator getStandardChangelogIterator(Contexts contexts, LabelExpression labelExpression,
                                                             DatabaseChangeLog changeLog) throws DatabaseException {
        return new ChangeLogIterator(changeLog,
                new ShouldRunChangeSetFilter(database),
                new ContextChangeSetFilter(contexts),
                new LabelChangeSetFilter(labelExpression),
                new DbmsChangeSetFilter(database),
                new IgnoreChangeSetFilter());
    }
```
When talk about the missing filters in the @return do you mean we should be adding them in the javadoc? It specifies a ChangelogIterator instance is returned and that's what the method is returning.

File: liquibase-standard/src/main/java/liquibase/hub/core/StandardHubService.java (hunk 1 above)
Comment (MalloD12) [reply liquibase-standard/src/main/java/liquibase/hub/core/StandardHubService.java:360]: I updated this branch with master and this file is gone. I'm also removing the `StandardHubServiceTest` class as well.

File: liquibase-standard/src/main/java/liquibase/util/StringUtil.java (hunk 3 above)
Comment (MalloD12) [reply liquibase-standard/src/main/java/liquibase/util/StringUtil.java:601]: Removed "css" and "copy code". Regarding <code> vs {@code} I replied in my previous comment. My understanding is they both has the same purpose but {@code} does not resolve indentation and line break issues, so maybe we can use any of them depending on each individual case or we can make an agreement and use one of them most of the times.

**Comment by MalloD12** [comment]
@XDelphiGrl I have made a few updates based on your comments. Can you please re-review them? And please let me know if you think there are more requested changes. 

Thanks,
Daniel.

File: liquibase-standard/src/main/java/liquibase/Liquibase.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (XDelphiGrl) [reply liquibase-standard/src/main/java/liquibase/Liquibase.java]: Thank you, @MalloD12 ! I misread the code for **@return**. I appreciate you removing `collectAllReasons` **@param**.

File: liquibase-standard/src/main/java/liquibase/hub/core/StandardHubService.java (hunk 1 above)
Comment (XDelphiGrl) [reply liquibase-standard/src/main/java/liquibase/hub/core/StandardHubService.java]: Thanks!

File: liquibase-standard/src/main/java/liquibase/util/StringUtil.java (hunk 3 above)
Comment (XDelphiGrl) [reply liquibase-standard/src/main/java/liquibase/util/StringUtil.java]: :) Thanks for getting rid of the "css" and "copy code" lines. If both `<code>` and `{@code}` perform the same function, I don't mind which we use.

**Comment by XDelphiGrl** [comment]
Thanks for the code changes, @MalloD12. I'm going to wait for the functional tests to run (since there is a lot of change due to the Hub refactoring work). When those pass, I'll do the final review/approval.

**Comment by XDelphiGrl** [comment]
For whatever reason, the test harness is lingering in the "pending" state. I've kicked off a manual build here: [https://github.com/liquibase/liquibase-test-harness/actions/runs/4996652294](https://github.com/liquibase/liquibase-test-harness/actions/runs/4996652294). 

I excluded the HSQL and Firebird databases as they are failing due to Titan errors.

**Review by XDelphiGrl — APPROVED** [review]
This is a JavaDoc cleanup PR that improves `@param` and `@return` documentation for many methods. 
* No serializable annotations are modified in this PR (which is good!)
* Functional tests and test harness executions are passing.
* No additional testing required.

APPROVED

**Pull request merged**